File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode167.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function twoSum(numbers, target) {
    var start = 0;
    var end = numbers.length - 1;
    var res = [];
    while (start <= end) {
        if (numbers[start] + numbers[end] == target) {
            res.push(start + 1);
            res.push(end + 1);
            return res;
        }
        else if (numbers[start] + numbers[end] > target) {
            end--;
        }
        else {
            start++;
        }
    }
    return res;
}
var arr = [2, 7, 11, 15];
var target = 9;
console.log(twoSum(arr, target));

File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode167.java
import java.util.*;

public class Leetcode167 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the size of the array: ");
        int size = input.nextInt();
        int[] arr = new int[size];
        for (int i = 0; i < size; i++) {
            System.out.printf("Enter %d number element of the array: ", i + 1);
            arr[i] = input.nextInt();
        }
        System.out.print("Enter target element: ");
        int target = input.nextInt();
        input.close();
        System.out.println(Arrays.toString(twoSum(arr, target)));
    }

    // A brute force solution
    private static int[] twoSum(int[] numbers, int target) {
        int start = 0, end = numbers.length - 1;
        while (start <= end) {
            if (numbers[start] + numbers[end] == target) {
                return new int[] { start + 1, end + 1 };
            } else if (numbers[start] + numbers[end] > target) {
                end--;
            } else {
                start++;
            }
        }
        return null;
    }
}
